bugsnag-android-unity/src/main/java/com/bugsnag/android/unity/BugsnagUnity.java

```diff
@@ -25,8 +25,14 @@ public boolean onError(Event event) {
                 Error error = event.getErrors().get(0);
                 String errorClass = error.getErrorClass();
                 String message = error.getErrorMessage();
-                return discardedEventErrorClass.equals(errorClass) &&
-                    (message == null || !pattern.matcher(message).find());
+                if(discardedEventErrorClass.equals(errorClass))
+                {
+                    if(message == null || !pattern.matcher(message).find())
+                    {
+                        return false;
+                    }
+                }
```

Comment:
```suggestion
                if(discardedEventErrorClass.equals(errorClass)) {
                    return message == null || !pattern.matcher(message).find();
                }
```

style nit: Java is typically same-line for brackets.